### DISCUSSION TITLE
Running Different Code Threads with Tabsets in Quarto?

### DISCUSSION THREAD

**Discussion by Oren-T** [quarto-dev/quarto-cli#5928] [Q&A]
### Description

Hi! I'm hoping to use Quarto to create tabsets for two different R packages, to show the side-by-side code on a published webpage (so rendering to HTML). However, these packages represent data tables as different types internally, and so when the second package runs, it overwrites the variable names with the new data types, which can lose compatibility with future methods. Is there any way for the two tabsets, if they are grouped, to essentially run in separate variable environments?

**Comment by cscheid** [comment]
You're asking a question that's a bit more general than tabsets; the same question could be asked about two consecutive code cells. I think the answer is to use different variable names (it would probably also help your documentation to have different data tables stored under different variable names so the user knows). You can create separate  [R environments](http://adv-r.had.co.nz/Environments.html#:~:text=To%20create%20an%20environment%20manually,env()%20.) if you want to write generic functions later.

**Comment by mcanouil** [comment]
Your question is not related to Quarto.
It is about R packages conflicts, there are various ways (explained on the Internet) such as using package prefix, using `callr` to run R code in subprocess, etc.
